src/main/java/at/porscheinformatik/sonarqube/licensecheck/gradle/GradleInvoker.java

```diff
@@ -0,0 +1,147 @@
+package at.porscheinformatik.sonarqube.licensecheck.gradle;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Options;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+class GradleInvoker {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GradleInvoker.class);
+
+    private final static String BUILD_GRADLE = "build.gradle";
+    private static String GRADLE_EXEC;
+
+    private final File projectRoot;
+
+    private String userSettings;
+
+    // todo: use gradle tooling api if possible
+    GradleInvoker(String projectRoot) throws Exception {
+        this.projectRoot = new File(projectRoot);
+        File buildGradle = new File(projectRoot, BUILD_GRADLE);
+
+        if (!buildGradle.exists()) {
+            throw new Exception("no build.gradle found");
+        }
+
+        GRADLE_EXEC = resolveGradleExecutable(this.projectRoot);
+
+        userSettings = null;
+        CommandLine cmd = getCommandLineArgs();
+        if (cmd != null) {
+            if (cmd.hasOption("I")) {
+                userSettings = cmd.getOptionValue("I");
+            }
+        }
+    }
+
+    String invoke(String... gradleTasks) throws IOException, GradleInvokerException {
+
+        String[] command = resolveFullGradleCommand(gradleTasks);
+
+        ProcessBuilder processBuilder = new ProcessBuilder();
+        Process process = processBuilder.command(command).directory(projectRoot).start();
+
+        InputStream errorStream = process.getErrorStream();
+        InputStream inputStream = process.getInputStream();
+        String stderr = getOutput(errorStream);
+        String stdout = getOutput(inputStream);
+
+        while (process.isAlive()) {
```

Comment:
Use Process#waitFor instead of busy wait and you will also get the exit value:
```suggestion
        if (process.waitFor() != 0) {
```